proxy.go

```diff
@@ -278,9 +275,12 @@ type ActionCandidate struct {
 	Operation ServiceOperation
 }
 
-func handleAWSRequest(req *http.Request, body []byte, respCode int) {
+func handleAWSRequest(resp *http.Response) {
+	req := resp.Request
 	host := req.Host
-	uri := req.RequestURI
+
+	body, _ := ioutil.ReadAll(req.Body)
+	req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
I moved the request body draining into this `handleAWSRequest` function to more closely match e.g. what `httputil.DumpRequest` does.